src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -1261,27 +1301,40 @@ public async Task GetResponseAsync_ResourceNotFound_ThrowsWebException()
                 $"HTTP/1.1 404 Not Found\r\nContent-Length: 0\r\n\r\n"));
         }
 
-        [Theory, MemberData(nameof(EchoServers))]
-        public async Task HaveResponse_GetResponseAsync_ExpectTrue(Uri remoteServer)
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task HaveResponse_GetResponseAsync_ExpectTrue(bool useSsl)
         {
-            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
-            using (WebResponse response = await request.GetResponseAsync())
+            var options = new LoopbackServer.Options { UseSsl = useSsl };
+
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
             {
+                HttpWebRequest request = WebRequest.CreateHttp(uri);
+                request.ServerCertificateValidationCallback = (a, b, c, d) => true;
+                request.UseDefaultCredentials = true;
+
+                using WebResponse response = await request.GetResponseAsync();
                 Assert.True(request.HaveResponse);
-            }
+            }, server => server.HandleRequestAsync(), options);
         }
 
         [Theory]
-        [MemberData(nameof(EchoServers))]
-        public async Task Headers_GetResponseHeaders_ContainsExpectedValue(Uri remoteServer)
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task Headers_GetResponseHeaders_ContainsExpectedValue(bool useSsl)
         {
-            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
-            using (HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync())
+            var options = new LoopbackServer.Options { UseSsl = useSsl };
+
+            const string HeadersPartialContent = "application/json";
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
             {
+                HttpWebRequest request = WebRequest.CreateHttp(uri);
+                request.ServerCertificateValidationCallback = (a, b, c, d) => true;
+                using WebResponse response = await request.GetResponseAsync();
                 string headersString = response.Headers.ToString();
-                string headersPartialContent = "Content-Type: application/json";
-                Assert.True(headersString.Contains(headersPartialContent));
-            }
+                Assert.Equal(HeadersPartialContent, response.Headers[HttpResponseHeader.ContentType]);
+            }, server => server.HandleRequestAsync(headers: new HttpHeaderData[] { new HttpHeaderData("Content-Type", HeadersPartialContent) }), options);
```

Comment:
Same echo server issue.  It seems like we should add a server.Echo() method that functionally does the same thing the remote echo server was doing.